### ISSUE TITLE
OpenCL not (directly) available on Mac

### ISSUE THREAD

**Issue by 0ge** [darktable-org/darktable#6257]
**Describe the bug**
When launching darktable the first time (also after updates), it is not possible to enable OpenCL in Preferences.

**To Reproduce**
1. Download latest version of DarkTable and copy to /Applications/
2. Go to Preferences -> cpu / gpu / memory
3. Check the 'activate OpenCL support' checkbox

**Expected behavior**
OpenCL is activated.

**Actual behavior**
The checkbox is disabled.

**Platform:**
 - Darktable Version: 3.2.1 (also experienced on some earlier version, 3.x)
 - OS: macOS 10.15.6
 - Intel Iris Graphics 6100 1536 MB

**Additional context**
It is possible to enable OpenCL if darktable is run from the command line. This only has to be done once, for every update.

I failed to capture the output on the first run (i.e. the one where OpenCL is actually enabled). Replacing the application in the /Applications-folder does not reproduce the issue, neither does running the application from another location.

```
~ /Applications/darktable.app/Contents/MacOS/darktable -d opencl

(process:855): GLib-GObject-CRITICAL **: 05:53:50.922: g_object_set: assertion 'G_IS_OBJECT (object)' failed
[dt_pthread_create] info: bumping pthread's stacksize from 524288 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 524288 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 524288 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 524288 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 524288 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 524288 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 524288 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 524288 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 524288 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 524288 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 524288 to 2097152
[dt_pthread_create] info: bumping pthread's stacksize from 524288 to 2097152
0.118487 [opencl_init] opencl related configuration options:
0.118522 [opencl_init] 
0.118525 [opencl_init] opencl: 1
0.118527 [opencl_init] opencl_scheduling_profile: 'default'
0.118529 [opencl_init] opencl_library: ''
0.118532 [opencl_init] opencl_memory_requirement: 768
0.118535 [opencl_init] opencl_memory_headroom: 400
0.118537 [opencl_init] opencl_device_priority: '*/!0,*/*/*'
0.118539 [opencl_init] opencl_mandatory_timeout: 200
0.118542 [opencl_init] opencl_size_roundup: 16
0.118544 [opencl_init] opencl_async_pixelpipe: 0
0.118546 [opencl_init] opencl_synch_cache: active module
0.118549 [opencl_init] opencl_number_event_handles: 25
0.118552 [opencl_init] opencl_micro_nap: 1000
0.118554 [opencl_init] opencl_use_pinned_memory: 0
0.118555 [opencl_init] opencl_use_cpu_devices: 0
0.118557 [opencl_init] opencl_avoid_atomics: 0
0.118558 [opencl_init] 
0.119267 [opencl_init] found opencl runtime library '/System/Library/Frameworks/OpenCL.framework/Versions/Current/OpenCL'
0.119313 [opencl_init] opencl library '/System/Library/Frameworks/OpenCL.framework/Versions/Current/OpenCL' found on your system and loaded
0.119320 [opencl_init] found 1 platform
0.125236 [opencl_init] found 2 devices
0.125291 [opencl_init] discarding CPU device 0 `Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz'.
0.125312 [opencl_init] device 1 `Intel(R) Iris(TM) Graphics 6100' supports image sizes of 16384 x 16384
0.125316 [opencl_init] device 1 `Intel(R) Iris(TM) Graphics 6100' allows GPU memory allocations of up to 384MB
[opencl_init] device 1: Intel(R) Iris(TM) Graphics 6100 
     GLOBAL_MEM_SIZE:          1536MB
     MAX_WORK_GROUP_SIZE:      256
     MAX_WORK_ITEM_DIMENSIONS: 3
     MAX_WORK_ITEM_SIZES:      [ 256 256 256 ]
     DRIVER_VERSION:           1.2(Jul  6 2020 11:56:20)
     DEVICE_VERSION:           OpenCL 1.2 
0.127879 [opencl_init] options for OpenCL compiler: -w  -DUNKNOWN=1 -I/Applications/darktable.app/Contents/Resources/share/darktable/kernels
0.129181 [opencl_init] compiling program `demosaic_ppg.cl' ..
0.130066 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/demosaic_ppg.cl.bin' MD5: 'f3bb318cb439803e3004eb41886bb25d' 
0.130083 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/demosaic_ppg.cl' MD5: 'f3bb318cb439803e3004eb41886bb25d'
0.130371 [opencl_build_program] successfully built program
0.130384 [opencl_build_program] BUILD STATUS: 0
0.130388 BUILD LOG:
0.130390 
0.130402 [opencl_init] compiling program `atrous.cl' ..
0.131093 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/atrous.cl.bin' MD5: '1bd10a37004fec374f85c217d478d1f7' 
0.131106 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/atrous.cl' MD5: '1bd10a37004fec374f85c217d478d1f7'
0.131231 [opencl_build_program] successfully built program
0.131239 [opencl_build_program] BUILD STATUS: 0
0.131243 BUILD LOG:
0.131246 
0.131259 [opencl_init] compiling program `basic.cl' ..
0.133034 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/basic.cl.bin' MD5: '113cfa341d7d421db543921fe00cdab2' 
0.133053 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/basic.cl' MD5: '113cfa341d7d421db543921fe00cdab2'
0.133422 [opencl_build_program] successfully built program
0.133433 [opencl_build_program] BUILD STATUS: 0
0.133437 BUILD LOG:
0.133438 
0.133449 [opencl_init] compiling program `blendop.cl' ..
0.134893 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/blendop.cl.bin' MD5: '3e1e559292238eb9b52318c6ba0a3768' 
0.134909 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/blendop.cl' MD5: '3e1e559292238eb9b52318c6ba0a3768'
0.135119 [opencl_build_program] successfully built program
0.135129 [opencl_build_program] BUILD STATUS: 0
0.135134 BUILD LOG:
0.135136 
0.135150 [opencl_init] compiling program `highpass.cl' ..
0.135832 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/highpass.cl.bin' MD5: '5374cfd8a91a52a25434bb1ff6af7f23' 
0.135846 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/highpass.cl' MD5: '5374cfd8a91a52a25434bb1ff6af7f23'
0.135954 [opencl_build_program] successfully built program
0.135962 [opencl_build_program] BUILD STATUS: 0
0.135966 BUILD LOG:
0.135969 
0.135981 [opencl_init] compiling program `nlmeans.cl' ..
0.136688 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/nlmeans.cl.bin' MD5: 'fde7ff3440e95ec8dbfb7a2f456ca80c' 
0.136704 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/nlmeans.cl' MD5: 'fde7ff3440e95ec8dbfb7a2f456ca80c'
0.136841 [opencl_build_program] successfully built program
0.136849 [opencl_build_program] BUILD STATUS: 0
0.136854 BUILD LOG:
0.136857 
0.136869 [opencl_init] compiling program `gaussian.cl' ..
0.137579 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/gaussian.cl.bin' MD5: '859287c6c084e56bd0802745b77deb93' 
0.137593 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/gaussian.cl' MD5: '859287c6c084e56bd0802745b77deb93'
0.137713 [opencl_build_program] successfully built program
0.137721 [opencl_build_program] BUILD STATUS: 0
0.137726 BUILD LOG:
0.137728 
0.137740 [opencl_init] compiling program `sharpen.cl' ..
0.138337 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/sharpen.cl.bin' MD5: '82677d23129088a99c04290aedefb2ca' 
0.138349 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/sharpen.cl' MD5: '82677d23129088a99c04290aedefb2ca'
0.138451 [opencl_build_program] successfully built program
0.138458 [opencl_build_program] BUILD STATUS: 0
0.138463 BUILD LOG:
0.138465 
0.138478 [opencl_init] compiling program `extended.cl' ..
0.139428 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/extended.cl.bin' MD5: '87b31991ff55c69fc1a5081640a587ee' 
0.139444 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/extended.cl' MD5: '87b31991ff55c69fc1a5081640a587ee'
0.139682 [opencl_build_program] successfully built program
0.139692 [opencl_build_program] BUILD STATUS: 0
0.139697 BUILD LOG:
0.139699 
0.139713 [opencl_init] compiling program `soften.cl' ..
0.140346 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/soften.cl.bin' MD5: 'b3446598016713ec70149ad155ef7fbc' 
0.140358 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/soften.cl' MD5: 'b3446598016713ec70149ad155ef7fbc'
0.140455 [opencl_build_program] successfully built program
0.140462 [opencl_build_program] BUILD STATUS: 0
0.140466 BUILD LOG:
0.140469 
0.140481 [opencl_init] compiling program `bilateral.cl' ..
0.141210 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/bilateral.cl.bin' MD5: 'ac66ff39ad590f724d3d66f53e8c663e' 
0.141224 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/bilateral.cl' MD5: 'ac66ff39ad590f724d3d66f53e8c663e'
0.141337 [opencl_build_program] successfully built program
0.141344 [opencl_build_program] BUILD STATUS: 0
0.141349 BUILD LOG:
0.141351 
0.141364 [opencl_init] compiling program `denoiseprofile.cl' ..
0.142075 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/denoiseprofile.cl.bin' MD5: '9a36a12874fbdd50daa91afb8725711b' 
0.142090 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/denoiseprofile.cl' MD5: '9a36a12874fbdd50daa91afb8725711b'
0.142257 [opencl_build_program] successfully built program
0.142266 [opencl_build_program] BUILD STATUS: 0
0.142271 BUILD LOG:
0.142273 
0.142286 [opencl_init] compiling program `bloom.cl' ..
0.142905 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/bloom.cl.bin' MD5: 'f5627328c3f3addb9f226ea85b1797e8' 
0.142918 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/bloom.cl' MD5: 'f5627328c3f3addb9f226ea85b1797e8'
0.143023 [opencl_build_program] successfully built program
0.143031 [opencl_build_program] BUILD STATUS: 0
0.143035 BUILD LOG:
0.143038 
0.143050 [opencl_init] compiling program `colorreconstruction.cl' ..
0.143691 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/colorreconstruction.cl.bin' MD5: '5e9563e616b6deca991a5a950b570f97' 
0.143701 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/colorreconstruction.cl' MD5: '5e9563e616b6deca991a5a950b570f97'
0.143798 [opencl_build_program] successfully built program
0.143806 [opencl_build_program] BUILD STATUS: 0
0.143810 BUILD LOG:
0.143812 
0.143821 [opencl_init] compiling program `demosaic_other.cl' ..
0.144370 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/demosaic_other.cl.bin' MD5: 'ff16e447236891e9b3063faa67ff7fc1' 
0.144380 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/demosaic_other.cl' MD5: 'ff16e447236891e9b3063faa67ff7fc1'
0.144470 [opencl_build_program] successfully built program
0.144476 [opencl_build_program] BUILD STATUS: 0
0.144478 BUILD LOG:
0.144480 
0.144491 [opencl_init] compiling program `demosaic_vng.cl' ..
0.145150 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/demosaic_vng.cl.bin' MD5: '2f842766e746f586d8d496edf6eed88f' 
0.145160 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/demosaic_vng.cl' MD5: '2f842766e746f586d8d496edf6eed88f'
0.145275 [opencl_build_program] successfully built program
0.145286 [opencl_build_program] BUILD STATUS: 0
0.145291 BUILD LOG:
0.145294 
0.145306 [opencl_init] compiling program `demosaic_markesteijn.cl' ..
0.146190 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/demosaic_markesteijn.cl.bin' MD5: '0f817d487453a1bc8c4a285dedd863b5' 
0.146201 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/demosaic_markesteijn.cl' MD5: '0f817d487453a1bc8c4a285dedd863b5'
0.146428 [opencl_build_program] successfully built program
0.146437 [opencl_build_program] BUILD STATUS: 0
0.146441 BUILD LOG:
0.146442 
0.146451 [opencl_init] compiling program `liquify.cl' ..
0.147051 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/liquify.cl.bin' MD5: 'c7a1f4fb0758fa48cbbf493834b6a1a9' 
0.147061 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/liquify.cl' MD5: 'c7a1f4fb0758fa48cbbf493834b6a1a9'
0.147147 [opencl_build_program] successfully built program
0.147154 [opencl_build_program] BUILD STATUS: 0
0.147159 BUILD LOG:
0.147162 
0.147171 [opencl_init] compiling program `basecurve.cl' ..
0.147839 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/basecurve.cl.bin' MD5: '9936f626cc351eec15a33a8b3fc3abb6' 
0.147849 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/basecurve.cl' MD5: '9936f626cc351eec15a33a8b3fc3abb6'
0.148046 [opencl_build_program] successfully built program
0.148055 [opencl_build_program] BUILD STATUS: 0
0.148058 BUILD LOG:
0.148060 
0.148069 [opencl_init] compiling program `locallaplacian.cl' ..
0.148951 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/locallaplacian.cl.bin' MD5: '0088f024a6a8752f2fb84fd2b23f7d63' 
0.148966 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/locallaplacian.cl' MD5: '0088f024a6a8752f2fb84fd2b23f7d63'
0.149094 [opencl_build_program] successfully built program
0.149103 [opencl_build_program] BUILD STATUS: 0
0.149108 BUILD LOG:
0.149110 
0.149123 [opencl_init] compiling program `dwt.cl' ..
0.149748 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/dwt.cl.bin' MD5: '44834bc88cd888550d003b90399530f3' 
0.149761 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/dwt.cl' MD5: '44834bc88cd888550d003b90399530f3'
0.149883 [opencl_build_program] successfully built program
0.149890 [opencl_build_program] BUILD STATUS: 0
0.149893 BUILD LOG:
0.149895 
0.149907 [opencl_init] compiling program `retouch.cl' ..
0.150500 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/retouch.cl.bin' MD5: 'ac1c2c55b9347affd015890fdc6a5f25' 
0.150508 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/retouch.cl' MD5: 'ac1c2c55b9347affd015890fdc6a5f25'
0.150607 [opencl_build_program] successfully built program
0.150612 [opencl_build_program] BUILD STATUS: 0
0.150615 BUILD LOG:
0.150616 
0.150624 [opencl_init] compiling program `filmic.cl' ..
0.151946 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/filmic.cl.bin' MD5: '5c6b2706e36078adc45831154f26c3cb' 
0.151957 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/filmic.cl' MD5: '5c6b2706e36078adc45831154f26c3cb'
0.152318 [opencl_build_program] successfully built program
0.152325 [opencl_build_program] BUILD STATUS: 0
0.152328 BUILD LOG:
0.152330 
0.152340 [opencl_init] compiling program `colorspaces.cl' ..
0.152749 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/colorspaces.cl.bin' MD5: '9e097c8517628b4e8d75f0178fcfa3dd' 
0.152757 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/colorspaces.cl' MD5: '9e097c8517628b4e8d75f0178fcfa3dd'
0.152837 [opencl_build_program] successfully built program
0.152842 [opencl_build_program] BUILD STATUS: 0
0.152845 BUILD LOG:
0.152846 
0.152854 [opencl_init] compiling program `basicadj.cl' ..
0.153433 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/basicadj.cl.bin' MD5: '214eb649c9cb4c526b8b6db996801524' 
0.153444 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/basicadj.cl' MD5: '214eb649c9cb4c526b8b6db996801524'
0.153627 [opencl_build_program] successfully built program
0.153634 [opencl_build_program] BUILD STATUS: 0
0.153639 BUILD LOG:
0.153641 
0.153654 [opencl_init] compiling program `rgbcurve.cl' ..
0.154285 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/rgbcurve.cl.bin' MD5: 'd34c82ad83d3ae264e513d0ca580fded' 
0.154298 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/rgbcurve.cl' MD5: 'd34c82ad83d3ae264e513d0ca580fded'
0.154380 [opencl_build_program] successfully built program
0.154387 [opencl_build_program] BUILD STATUS: 0
0.154391 BUILD LOG:
0.154394 
0.154405 [opencl_init] compiling program `guided_filter.cl' ..
0.155044 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/guided_filter.cl.bin' MD5: 'b4043110273e87a6eec2aa1d2e6afe72' 
0.155054 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/guided_filter.cl' MD5: 'b4043110273e87a6eec2aa1d2e6afe72'
0.155169 [opencl_build_program] successfully built program
0.155179 [opencl_build_program] BUILD STATUS: 0
0.155184 BUILD LOG:
0.155186 
0.155197 [opencl_init] compiling program `hazeremoval.cl' ..
0.155753 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/hazeremoval.cl.bin' MD5: 'd0331728ac5e9344dc7d5c801051d3c2' 
0.155763 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/hazeremoval.cl' MD5: 'd0331728ac5e9344dc7d5c801051d3c2'
0.155876 [opencl_build_program] successfully built program
0.155884 [opencl_build_program] BUILD STATUS: 0
0.155889 BUILD LOG:
0.155892 
0.155903 [opencl_init] compiling program `lut3d.cl' ..
0.156439 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/lut3d.cl.bin' MD5: '8eea5180529dd174daee4d6ecdc99643' 
0.156449 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/lut3d.cl' MD5: '8eea5180529dd174daee4d6ecdc99643'
0.156537 [opencl_build_program] successfully built program
0.156545 [opencl_build_program] BUILD STATUS: 0
0.156550 BUILD LOG:
0.156552 
0.156559 [opencl_init] compiling program `rgblevels.cl' ..
0.157032 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/rgblevels.cl.bin' MD5: '7881e4a335114a208e6c16dd51492e09' 
0.157041 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/rgblevels.cl' MD5: '7881e4a335114a208e6c16dd51492e09'
0.157135 [opencl_build_program] successfully built program
0.157142 [opencl_build_program] BUILD STATUS: 0
0.157146 BUILD LOG:
0.157148 
0.157159 [opencl_init] compiling program `negadoctor.cl' ..
0.157683 [opencl_load_program] loaded cached binary program from file '/Users/osau/.cache/darktable/cached_kernels_for_IntelRIrisTMGraphics6100_12Jul62020115620/negadoctor.cl.bin' MD5: '0a4ddf41212f9aa2c8314c3cadf0f980' 
0.157692 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/negadoctor.cl' MD5: '0a4ddf41212f9aa2c8314c3cadf0f980'
0.157766 [opencl_build_program] successfully built program
0.157772 [opencl_build_program] BUILD STATUS: 0
0.157776 BUILD LOG:
0.157779 
0.157788 [opencl_init] kernel loading time: 0.0289 
0.157799 [opencl_init] OpenCL successfully initialized.
0.157802 [opencl_init] here are the internal numbers and names of OpenCL devices available to darktable:
0.157804 [opencl_init]		0	'Intel(R) Iris(TM) Graphics 6100'
0.157806 [opencl_init] FINALLY: opencl is AVAILABLE on this system.
0.157808 [opencl_init] initial status of opencl enabled flag is ON.
0.157896 [opencl_create_kernel] successfully loaded kernel `blendop_mask_Lab' (0) for device 0
...
0.158903 [opencl_create_kernel] successfully loaded kernel `guided_filter_generate_result' (41) for device 0
0.158946 [opencl_priorities] these are your device priorities:
0.158950 [opencl_priorities] 		image	preview	export	thumbs	preview2
0.158952 [opencl_priorities]		0	-1	0	0	-1
0.158955 [opencl_priorities] show if opencl use is mandatory for a given pixelpipe:
0.158956 [opencl_priorities] 		image	preview	export	thumbs	preview2
0.158958 [opencl_priorities]		0	0	0	0	0
0.158963 [opencl_synchronization_timeout] synchronization timeout set to 200

(darktable:855): GLib-GObject-WARNING **: 05:53:51.264: invalid cast from 'GtkMenuBar' to 'GtkWindow'

(darktable:855): Gtk-CRITICAL **: 05:53:51.264: gtk_window_add_accel_group: assertion 'GTK_IS_WINDOW (window)' failed
0,958299 [opencl_create_kernel] successfully loaded kernel `colorbalance' (42) for device 0
...
2,343063 [opencl_create_kernel] successfully loaded kernel `denoiseprofile_reduce_second' (234) for device 0
```

**Comment by MStraeten** [comment]
See https://github.com/darktable-org/darktable/issues/2810
It needs a couple of attempts to compile the kernels. You can use a script to do this for you.

**Comment by parafin** [comment]
Closing as duplicate.

**Comment by 0ge** [comment]
Thanks! Just as a note; the behaviour I'm seeing is _maybe_ slightly different and I think I described it incorrectly. It seems it never tries to 'compile the kernels' when opened directly from the `darktable.app`. What I have experienced is that it is _never_ possible to enable OpenCL if you only open it from `darktable.app`. But when run from the command line it appears to work on the first (or maybe second) try. It could just be coincidence, of course.
